### ISSUE TITLE
wsb for Pitchers is not adjusting for playing time

### ISSUE THREAD

**Issue by skoormit** [danseguin23/ootp-calculator#9]
https://github.com/danseguin23/ootp-calculator/blob/3934d5ffa39bb673a82284aacede769f7cb80941/src/data-classes.js#L430

A wSB value is determined for each pitcher based on Hold rating, but is never adjusted for playing time.
As a result, all players with the same Hold rating have the same wSB value added to their runs allowed, regardless of projected playing time (or any other aspect of their projected performance).

I'm not sure what the formula should be, because I'm not sure what the wSB value itself represents.
Hold ratings (20-80 scale) of 20, 50, 80 give wSB of approx 1.1, 0, -0.6.
That's in the ballpark (heh) of the range of [Pitcher Stealing Runs provided by Baseball Savant for 2023](https://baseballsavant.mlb.com/leaderboard/pitcher-running-game) (-2 to 3, but the scale is opposite, so equivalent to wSB of 2 to -3). I presume your wSB slope and intercept values came from observed OOTP output, and I can certainly believe that the OOTP engine doesn't simulate this aspect of the game perfectly.

Perhaps a reasonable fix is just to multiply wSB by AB/550 (since your other calcs take 550 AB as the default for a full season). That gives less credit to good holders than they deserve IRL, but maybe it's about right for the OOTP engine.

**Comment by danseguin23** [comment]
Skoormit! Thanks for the comment and I apologize for any trauma that may have been inflicted as a result of having to look through my code. wSB is only used to calculate runs allowed and earned runs (line 436/437), which are later scaled down to playing time when assigning to `this` (line 452/453). So on its own, wSB isn’t actually shown or used directly.

**Comment by skoormit** [comment]
> Skoormit! Thanks for the comment and I apologize for any trauma that may have been inflicted as a result of having to look through my code. wSB is only used to calculate runs allowed and earned runs (line 436/437), which are later scaled down to playing time when assigning to `this` (line 452/453). So on its own, wSB isn’t actually shown or used directly.

Funny, I actually came to this realization while trying to fall asleep last night. Figured I'd come here first thing this morning and post a "nevermind, I'm dense," but you already read it.

So, I apologize for any trauma that may have been inflicted as a result of having to read my ham-handed attempt to parse your (perfectly cromulent) code.

Your code does a fine job, by the way. My confusion arose mostly from your convention of using very simple variable names. That convention has plenty of merit, though my own preference is to use (sometimes ridiculously) long variable names. I might have called `wsb` something like `wtdSBRunsPer550AB`.

More importantly than all that: thank you, thank you, thank you for this utility, and for sharing the source code. It has been a tremendous boon to my enjoyment of OOTP.

May the road rise to meet you, and may all your prospects develop.
